ui/org.eclipse.pde.ui/src/org/eclipse/pde/internal/ui/editor/ShowInPackageExplorer.java

```diff
@@ -0,0 +1,49 @@
+/*******************************************************************************
+ * Copyright (c) 2025 ArSysOp
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * https://www.eclipse.org/legal/epl-2.0/.
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *     Alexander Fedorov (ArSysOp) - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.pde.internal.ui.editor;
+
+import java.util.Objects;
+import java.util.Optional;
+
+import org.eclipse.jdt.core.IPackageFragment;
+import org.eclipse.jdt.ui.JavaUI;
+import org.eclipse.jface.viewers.StructuredSelection;
+import org.eclipse.pde.internal.ui.PDEPlugin;
+import org.eclipse.ui.IViewPart;
+import org.eclipse.ui.PartInitException;
+import org.eclipse.ui.part.IShowInTarget;
+import org.eclipse.ui.part.ShowInContext;
+
+public final class ShowInPackageExplorer implements Runnable {
```

Comment:
Wouldn't it be simpler to just have a static method `showInPackageExplorer(IPackageFragment fragment)`?
Then the callers would not first have to map an optional to a new instance of this class and run it if present, instead it could just be `...ifPresent(ShowInPackageExplorer::showInPackageExplorer)`.

And encapsulating this one action in a whole class is IMHO a bit heavy-weight. I think the static method would then be sufficiently small to add it e.g. to `EditorUtilities`, where many similar methods already reside.

Reply:
I fail to see what benefits it gives us to implement runnable here? Also the class does not seem to encapsulate any data except it being a data-carrier for the `IPackageFragment`. So using a static method in a place that already exits and could be reused by other place to not look "bad" to me, even though I think there are no strict rules we are forced to follow.

In general of course try to follow the existing style of a project (regardless of its considered bad or not) could have some like this results usually in smaller patch sets that are easier to review and the project has not to learn new styles.

So I think that can be sen more as a guidance or maybe a personal preference than some hard rules we like to enforce.

Reply:
Strange GitHub infra issue
![image](https://github.com/user-attachments/assets/f73decd7-dc44-4041-b156-bd26867d491c)

Reply:
did another force push to my fork and now it is updated